### PR TITLE
chore: adaptations for nightly-2024-09-23

### PR DESCRIPTION
**Pull request by jcommelin**

### CODE REVIEW

File: Mathlib/FieldTheory/KummerExtension.lean [hunk 1]
```diff
@@ -272,6 +272,11 @@ theorem Polynomial.separable_X_pow_sub_C_of_irreducible : (X ^ n - C a).Separabl
     AdjoinRoot.algebraMap_eq,
     X_pow_sub_C_eq_prod (hζ.map_of_injective (algebraMap K _).injective) hn
     (root_X_pow_sub_C_pow n a), separable_prod_X_sub_C_iff']
+  #adaptation_note
+  /--
+  After https://github.com/leanprover/lean4/pull/5376 we need to provide this helper instance.
+  -/
+  have : MonoidHomClass (K →+* K[n√a]) K K[n√a] := inferInstance
```
Comment (YaelDillies) [inline Mathlib/FieldTheory/KummerExtension.lean:279]: Sad

File: Mathlib/MeasureTheory/Measure/Content.lean [hunk 2]
```diff
@@ -310,7 +310,7 @@ variable [S : MeasurableSpace G] [BorelSpace G]
 
 /-- For the outer measure coming from a content, all Borel sets are measurable. -/
 theorem borel_le_caratheodory : S ≤ μ.outerMeasure.caratheodory := by
-  rw [@BorelSpace.measurable_eq G _ _]
+  rw [@BorelSpace.measurable_eq G _ _ _]
```
Comment (YaelDillies) [inline Mathlib/MeasureTheory/Measure/Content.lean:313]:
```suggestion
  rw [BorelSpace.measurable_eq (α := G)]
```
maybe?

File: Mathlib/Order/SuccPred/LinearLocallyFinite.lean [hunk 3]
```diff
@@ -194,12 +194,13 @@ variable [SuccOrder ι] [IsSuccArchimedean ι] [PredOrder ι] {i0 i : ι}
 the range of `toZ`. -/
 def toZ (i0 i : ι) : ℤ :=
   dite (i0 ≤ i) (fun hi ↦ Nat.find (exists_succ_iterate_of_le hi)) fun hi ↦
-    -Nat.find (exists_pred_iterate_of_le (not_le.mp hi).le)
+    -Nat.find (@exists_pred_iterate_of_le _ _ P _ _ _ (not_le.mp hi).le)
```
Comment (YaelDillies) [inline Mathlib/Order/SuccPred/LinearLocallyFinite.lean:197]: What's going on here?

File: Mathlib/Topology/Order/LawsonTopology.lean [hunk 4]
```diff
@@ -144,7 +144,7 @@ instance instIsLawson : IsLawson (WithLawson α) := ⟨rfl⟩
 /-- If `α` is equipped with the Lawson topology, then it is homeomorphic to `WithLawson α`.
 -/
 def homeomorph [TopologicalSpace α] [IsLawson α] : WithLawson α ≃ₜ α :=
-  ofLawson.toHomeomorphOfInducing ⟨by erw [@IsLawson.topology_eq_lawson α _ _, induced_id]; rfl⟩
+  ofLawson.toHomeomorphOfInducing ⟨by erw [@IsLawson.topology_eq_lawson α _ _ _, induced_id]; rfl⟩
```
Comment (YaelDillies) [inline Mathlib/Topology/Order/LawsonTopology.lean:147]:
```suggestion
  ofLawson.toHomeomorphOfInducing ⟨by erw [IsLawson.topology_eq_lawson (α := α), induced_id]; rfl⟩
```

**Review by YaelDillies — COMMENTED** [review]
All the `HomClass` business looks very reasonable, except for that one regression

File: Mathlib/Order/SuccPred/LinearLocallyFinite.lean (hunk 3 above)
Comment (kim-em) [reply Mathlib/Order/SuccPred/LinearLocallyFinite.lean:197]: Couldn't work it out... :-(

Comment (YaelDillies) [reply Mathlib/Order/SuccPred/LinearLocallyFinite.lean]: Ah, that looks better! Must be some change in elaboration

File: Mathlib/GroupTheory/GroupAction/Quotient.lean [hunk 5]
```diff
@@ -404,6 +404,7 @@ noncomputable def equivSubgroupOrbitsQuotientGroup [IsPretransitive α β]
     convert one_mem H
     rw [inv_mul_eq_one, eq_comm, ← inv_mul_eq_one, ← Subgroup.mem_bot, ← free (g⁻¹ • x),
         mem_stabilizer_iff, mul_smul, (exists_smul_eq α (g⁻¹ • x) x).choose_spec]
+    infer_instance
```
Comment (jcommelin) [inline Mathlib/GroupTheory/GroupAction/Quotient.lean:407]: Should this one also come with an adaptation note?
```suggestion
  #adaptation_note
  /--
  After https://github.com/leanprover/lean4/pull/5376 we need to search for this instance explicitly.
  -/
    infer_instance
```

Comment (kim-em) [reply Mathlib/GroupTheory/GroupAction/Quotient.lean:407]: More specifically, it's that `convert` isn't solving it by typeclass search anymore, but just spitting out an unsolved implicit argument.

If there's anything to do here, it is to change the behaviour of `convert` to more agressively solve such goals with `infer_instance` itself.

Comment (jcommelin) [reply Mathlib/GroupTheory/GroupAction/Quotient.lean:407]:
```suggestion
    #adaptation_note
    /--
    After https://github.com/leanprover/lean4/pull/5376 we need to search for this instance explicitly.
    TODO: change the behaviour of `convert` to more agressively solve such goals with `infer_instance` itself.
    -/
    infer_instance
```

**Pull request merged**